README.md

```diff
@@ -629,6 +629,151 @@ Can you try updating the rendering function to also show `entry.content` and
 > You might need to use `switch` and the `dangerouslySetInnerHTML` prop for
 > `entry.content`, and `Array.map` for `entry.links`.
 
+## Step 8: Using input fields to get data for any user
+
+In this step, we'll enhance our app by letting users enter a GitHub username to
+fetch and display their activity feed.
+
+To do this, we will have to update our app in a few ways.
+
+First of all, we will add a couple more state hooks to store `username` and the
+`debounceTimeout`. For this, we will use `React.useState`, which we have used
+before.
+
+Then, we will add an input field to capture the GitHub username. We can use a
+label for clarity. We also have to ensure that the input value is bound to the
+`username` state:
+
+```reason
+<label htmlFor="username-input"> {React.string("Username:")} </label>
+<input
+  id="username-input"
+  value=username
+  onChange={event => {
+    setUsername(event->React.Event.Form.target##value);
+    setData(_ => Loading);
+  }}
+  placeholder="Enter GitHub username"
+/>
+```
+
+Finally, we have to modify our data fetching logic to get data for the current
+user. This will involve the following steps:
+
+- Remove the previous effect and move its content to a new function `fetchFeed`.
+  You'll need to modify the URL to dynamically pass the `username`, this can be
+  done by leveraging string concatenation operator `++` (see the [syntax
+  cheatsheet](https://reasonml.github.io/docs/en/syntax-cheatsheet)).
+- Create a new effect that has `username` as a dependency (hint: use
+  `React.useEffect1`)
+- Inside the effect, make sure to:
+    - clear the existing timeout `debounceTimeout` if it has a value set (its
+      value is `Some(_)`) using
+      [Js.Global.clearTimeout](https://melange.re/v4.0.0/api/re/melange/Js/Global/#val-clearTimeout)
+    - set a new timeout using
+      [Js.Global.setTimeout](https://melange.re/v4.0.0/api/re/melange/Js/Global/#val-setTimeout).
+      Let's set it to 500ms.
+    - store the new timeout id using `setDebounceTimeout` function from the
```

Comment:
Good idea. I will remove the debounce and use `onKeyDown` instead.

Reply:
Actually, removing this debounce leads to more complicated logic in the next step, because we have to account for more states that weren't there before, like the user is typing but hasn't pressed `Enter` yet. I am not sure which path is easier.

Reply:
Ended up keeping the `Enter` approach, see https://github.com/ahrefs/react-alicante-workshop/pull/6/commits/1e63afcbe33c61d6e38e696b507d10400a80f14b.